sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -174,8 +174,6 @@ class JDBCRDD(
     sparkContext,
     name = "JDBC query execution time")
 
-  private lazy val dialect = JdbcDialects.get(url)
```

Comment:
@urosstan-db Seems like recreating this causing an issue. Mind taking a look?

Reply:
It's from https://github.com/apache/spark/commit/e6a3385e27fa95391433ea02fa053540fe101d40

Reply:
@HyukjinKwon Before PR you referenced, dialect was created also by resolution of URL, in `compute` method of `JDBCRDD`, on executor side.

Reply:
I am ok with this change. Serialization of dialect should not affect performances

Reply:
Thanks @HyukjinKwon and @urosstan-db ! I've updated the PR with the proposed diff. Would you like me to squash commits?